packages/ra-ui-materialui/src/input/ArrayInput/ArrayInput.spec.tsx

```diff
@@ -264,6 +264,53 @@ describe('<ArrayInput />', () => {
         expect(screen.queryByText('test helper text')).not.toBeNull();
     });
 
+    it('should update the form state to dirty, and allow submit, on updating an array input with default value', async () => {
+        const TestArrayInputs = () => {
+            const { reset } = useFormContext();
+
+            // mimic's the scaffolding elsewhere in the app but does not disable the form
+            // without this (before array input side effects)
+            React.useEffect(() => {
+                reset();
+            }, [reset]);
+
+            return (
+                <ArrayInput
+                    resource="bar"
+                    source="arr"
+                    defaultValue={[{ id: 'foo' }]}
+                >
+                    <SimpleFormIterator>
+                        <TextInput source="id" />
+                    </SimpleFormIterator>
+                </ArrayInput>
+            );
+        };
+
+        render(
+            <AdminContext dataProvider={testDataProvider()}>
+                <SimpleForm onSubmit={jest.fn}>
+                    <TestArrayInputs />
+                </SimpleForm>
+            </AdminContext>
+        );
+        const submitButton = screen
+            .getByLabelText('ra.action.save')
+            .closest('button');
+
+        await waitFor(() => {
+            expect(submitButton?.disabled).toBe(true);
+        });
```

Comment:
Another weirdness is that getting the submit button using the preferred (as recommended by the react testing team) `getByRole` followed by `expect(...button).isDisabled()` does not work even when the logged out element does show the disabled property as truthy?

Reply:
I don't know why this is not working, but your solution is good enough IMO